codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/customize/RustCodegenDecorator.kt

```diff
@@ -68,6 +69,8 @@ interface RustCodegenDecorator<C : CoreCodegenContext> {
         currentProtocols
 
     fun transformModel(service: ServiceShape, model: Model): Model = model
+
+    fun supportsCodegenContext(clazz: Class<*>): Boolean
```

Comment:
If you want to make this slightly simpler, you can add a `val codegenContextClass: Class<*>` after `val order`, and then you can provide a default implementation of `supportsCodegenContext` with the actual check in it. Then implementations only have to set the class value rather than implementing the function.

Should we use a stricter generic bound than `Class<*>`? It will always implement `CoreCodegenContext`, right?

Reply:
> If you want to make this slightly simpler, you can add a val codegenContextClass: Class<*> after val order, and then you can provide a default implementation of supportsCodegenContext with the actual check in it. Then implementations only have to set the class value rather than implementing the function.

That would make `val codegenContextClass: Class<*>` part of the public interface, at which point `supportsCodegenContext` is no longer needed and `fromClasspath` can just access the property.

> Should we use a stricter generic bound than Class<*>? It will always implement CoreCodegenContext, right?

Yes. But adding a stricter bound would entail making `RustCodegenDecorator` generic over another type parameter, so that we can write `U: CoreCodegenContext`, `supportsCodegenContext(clazz: Class<U>)`. Not worth it. The bound is already enforced in the signature of the callsite anyway.

Reply:
You should be able to restrict it using `Class<out CoreCodegenContext>` without introducing another generic.

Reply:
Smart! https://github.com/awslabs/smithy-rs/pull/1592/commits/e3f498c2b204e28cf5667ee7f3fc9f0ee1a6101b